cs-minibot-platform-src/src/main/java/examples/gobot/AIUtil.java

```diff
@@ -0,0 +1,371 @@
+package examples.gobot;
+
+import basestation.BaseStation;
+import basestation.vision.VisionCoordinate;
+import basestation.vision.VisionObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ */
+public class AIUtil {
+
+    private final int numLines;
+    private final double min;
+    private final double max;
+    Course course;
+
+    public AIUtil(int numLines, double min, double max) {
+        this.numLines = numLines;
+        this.min = min;
+        this.max = max;
+    }
+
+    /**
+     *
+     * @param e1 Equation one
+     * @param e2 Equation two
+     * @return Point representing the intersection between e1 and e2
+     */
+    public Point intersection(Equation e1, Equation e2){
+        try{
+            /*
+            Two vertical lines should not intersection each other
+            unless they are the same line
+             */
+            if (e1.vertical && e2.vertical){
+                return null;
+            }
+            //If either are vertical, just plug in
+            else if (e1.vertical || e2.vertical) {
+                if (e1.vertical){
+                    return new Point(e1.xval, e1.xval * e2.slope +
+                            e2.yIntercept);
+                }
+                else{
+                    return new Point(e2.xval, e2.xval * e1.slope +
+                            e1.yIntercept);
+                }
+
+            }
+            else {
+                double x = (e2.yIntercept - e1.yIntercept)/(e1.slope -
+                        e2.slope);
+                double y = e1.slope * x + e1.yIntercept;
+                return new Point(x, y);
+            }
+        }
+
+        //no intersection
+        catch (Exception noIntersection) {
+            noIntersection.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     *
+     * @param cors arraylist of VisionCoordinates of the track
+     * @return arraylist of all equations found between each pair of
```

Comment:
don't need to fix this, but as general advice, you should return List instead of ArrayList as it allows for implementations to be swapped out in the future if someone decides to optimize the code.